lib/async_hooks.js

```diff
@@ -461,6 +462,19 @@ function init(asyncId, type, triggerId, resource) {
   processing_hook = false;
 }
 
+const JSProviders = new Set();
+
+function registerTypeName(type) {
+  if (JSProviders.has(type))
+    throw new errors.TypeError('ERR_ASYNC_PROVIDER_NAME', type);
```

Comment:
here's some insanity, but could be useful for debugging:
```js
const JSProviders = new Map();
function registerTypeName(type) {
  if (JSProviders.has(type))
    throw new errors.TypeError('ERR_ASYNC_PROVIDER_NAME', type);  // include the old stack somehow
  JSProviders.set(type, new Error().stack);
}
```

Not suggesting we implement this, but thought something like it could be helpful for identifying the bugger who is using my type name.